models/ecoli/analysis/variant/doubling_time_histogram.py

```diff
@@ -1,132 +1,160 @@
+				only_successful=True)
 
-			all_cells = self.ap.get_cells(variant=[variant],
-										  only_successful=True)
 			if len(all_cells) == 0:
 				continue
 
+			# Get mask to exclude cells that timed out
 			exclude_timeout_cell_mask = stacked_cell_threshold_mask(
-				all_cells, 'Main', 'time', MAX_CELL_LENGTH,
+				all_cells, 'Main', 'time', max_cell_length,
 				fun=lambda x: (x[-1] - x[0]) / 60.).squeeze()
-			all_cells_gens = np.array([int(os.path.basename(os.path.dirname(
-				cell_path))[-6:]) for cell_path in all_cells])[exclude_timeout_cell_mask]
-			generations[variant] = all_cells_gens
 
-			# Doubling times
-			dt = read_stacked_columns(all_cells, 'Main', 'time',
-									  fun=lambda x: (x[-1] - x[0]) / 60.).squeeze()
-			doubling_times[variant] = dt[exclude_timeout_cell_mask]
+			# Get doubling times from cells with this variant index
+			dt = read_stacked_columns(
+				all_cells, 'Main', 'time',
+				fun=lambda x: (x[-1] - x[0]) / 60.).squeeze()
+			doubling_times[variant_index] = dt[exclude_timeout_cell_mask]
 
 			# Count the number of simulations that reach gen COUNT_INDEX + 1
-			num_count_gen = len(self.ap.get_cells(variant=[variant],
-							  generation = [COUNT_INDEX],
-												  only_successful=True))
-			num_zero_gen = len(self.ap.get_cells(variant=[variant],
-							  generation = [0], only_successful=True))
-			reached_count_gen[variant] = num_count_gen / num_zero_gen
-
-		# Plotting
-		std_xlim = [30,185]
-		std_dt_xlab = 'Doubling Time (min)'
-		std_bar_xlab = 'Variant'
-		std_bar_ylab = 'Percentage of Sims that Reached Generation ' \
-					   + str(COUNT_INDEX+1)
-
-		data_start = [0]
-		data_end = [MAX_CELL_INDEX]
-		plot_label = ['_all_gens']
-		if exclude_early_gens:
-			data_start += [0,MIN_LATE_CELL_INDEX]
-			data_end += [MIN_LATE_CELL_INDEX,MAX_CELL_INDEX]
-			plot_label += ['_early_gens', '_late_gens']
-
-		for j in range(len(data_start)):
-			_, axes = plt.subplots(2, 1, figsize=(10, 10))
-			labeled_indexable_hist(self, axes[0], doubling_times, generations,
-								   data_start[j], data_end[j], COLORS,
-								   std_dt_xlab)
-			self.bar(axes[1], reached_count_gen, std_bar_xlab, std_bar_ylab)
-			plt.tight_layout()
-			exportFigure(plt, plotOutDir, plotOutFileName+plot_label[j],
-						 metadata)
+			num_count_gen = len(
+				self.ap.get_cells(
+					variant=[variant_index], generation=[n_total_gens - 1],
+					only_successful=True)
+				)
+
+			# Count the number of simulations at generation zero
+			num_zero_gen = len(
+				self.ap.get_cells(
+					variant=[variant_index], generation=[0],
+					only_successful=True)
+				)
+
+			# Calculate completion rate
+			completion_rates[variant_index] = num_count_gen / num_zero_gen
+
+		n_variants = len(doubling_times)
+
+		fig = plt.figure(figsize=(8, 2*(n_variants + 1)))
+		bins = np.linspace(
+			DOUBLING_TIME_BOUNDS_MINUTES[0],
+			DOUBLING_TIME_BOUNDS_MINUTES[1],
+			N_BINS + 1
+			)
+
+		# First subplot shows all histograms for each variant
```

Comment:
For some reason, I feel like this comment implies there are multiple histograms for each variant? For clarity, I would consider changing it to something like `# First subplot overlays the histograms of all variants` to indicate that there is one histogram for each variant and the data for each is overlayed rather than merged